sim/druid/rake.go

```diff
@@ -0,0 +1,138 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+)
+
+type RakeRankInfo struct {
+	id            int32
+	level         int32
+	initialDamage float64
+	dotDamage     float64
+}
+
+var rakeSpells = []RakeRankInfo{
+	{
+		id:            1822,
+		level:         24,
+		initialDamage: 19.0,
+		dotDamage:     39.0,
+	},
+	{
+		id:            1823,
+		level:         34,
+		initialDamage: 29.0,
+		dotDamage:     57.0,
+	},
+	{
+
+		id:            1824,
+		level:         44,
+		initialDamage: 43.0,
+		dotDamage:     75.0,
+	},
+	{
+
+		id:            9904,
+		level:         54,
+		initialDamage: 58.0,
+		dotDamage:     96.0,
+	},
+}
+
+var rakeTicks = 3.0
+
+// SoD balance passive EFFECT1 and EFFECT2 mod for Rake
+// See https://www.wowhead.com/classic/spell=436895/s03-tuning-and-overrides-passive-druid
+// Mod Eff# should be base value only.
+var baseDmgMultiplier = 1.5
+
+func (druid *Druid) registerRakeSpell() {
+	// Add highest available rake rank for level.
+	for rank := len(rakeSpells) - 1; rank >= 0; rank-- {
+		if druid.Level >= rakeSpells[rank].level {
+			config := druid.newRakeSpellConfig(rakeSpells[rank])
+			druid.Rake = druid.RegisterSpell(Cat, config)
+			return
+		}
+	}
```

Comment:
Instead of iterating through the ranks like this, you could get the correct rank information from the start using a map of levels to ranks like this:
```
rankConfig := map[int32]int32{
  25: 1,
  40: 2,
  50: 3,
  60: 4,
}[druid.Level]
```

Reply:
I intentionally tried to avoid hardcoding levels so it's theoretically working with arbitrary levels, not just 25, 40, 50 and 60. I know that realistically those levels are the only ones ever used, but it still doesn't feel right to me to restrict it for no reason. Should I change it?

Reply:
Fair enough!